packages/react-router-dom/__tests__/navigate-encode-params-test.tsx

```diff
@@ -88,7 +89,8 @@ describe("navigate with params", () => {
       let pathname = window.location.pathname.replace(/%20/g, "+");
       expect(pathname).toEqual("/blog/react+router");
 
-      expect(node.innerHTML).toMatch(/react router/);
+      // Note decodeURIComponent doesn't decode +
+      expect(node.innerHTML).toMatch(/react\+router/);
```

Comment:
This was a false positive test previously - and was seemingly only because we weren't resetting history state between tests.  `decodeURIComponent` doesn't decode the `+` sign